### PR TITLE
Additional adjustments to the flow graph verifier.

### PR DESCRIPTION
**Pull request by 333fred**
@AlekseyTs @dotnet/roslyn-compiler for review. This should be the last change required for https://github.com/dotnet/roslyn/issues/31450.

@jaredpar, could you PR the IOperation test changes for CI to this branch so I can mark this as fixing that bug?

### CODE REVIEW

File: src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs [hunk 1]
```diff
@@ -1012,17 +1012,23 @@ bool isLongLivedCaptureReferenceSyntax(SyntaxNode captureReferenceSyntax)
...
+                                switch (syntax.Kind())
+                                {
+                                    case CSharp.SyntaxKind.ObjectCreationExpression:
+                                        if (((CSharp.Syntax.ObjectCreationExpressionSyntax)syntax).Initializer?.Expressions.Any() == true)
+                                        {
+                                            return true;
+                                        }
+                                        break;
+                                    default:
+                                        syntax = syntax.Parent;
```
Comment (AlekseyTs) [inline src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs]: >syntax = syntax.Parent [](start = 40, length = 22)

This looks too relaxed. Basically any syntax below CSharp.Syntax.ExpressionSyntax will return true. What scenario are we trying to handle?

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs (hunk 1 above)
Comment (333fred) [reply src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs]: We're trying to handle this scenario: [https://github.com/dotnet/roslyn/pull/31724/files#diff-44ef19e4c61837033de3e3b7f6002830R12798](https://github.com/dotnet/roslyn/pull/31724/files#diff-44ef19e4c61837033de3e3b7f6002830R12798). Where the property references uses the flow capture from a containing region.

---
In reply to: [241497757](https://github.com/dotnet/roslyn/pull/31724#discussion_r241497757) [](ancestors = 241497757)

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs [hunk 2]
```diff
@@ -12703,5 +12703,149 @@ static void F(A a)
...
+        Next (Regular) Block[B2]
+            Entering: {R2}
+
+    .locals {R2}
+    {
+        CaptureIds: [2]
+        Block[B2] - Block
+            Predecessors: [B1]
+            Statements (1)
+                IFlowCaptureOperation: 2 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'P2')
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs]: >IFlowCaptureOperation: 2 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'P2') [](start = 16, length = 91)

Is this the flow capture that we want to treat as long lived? Can we detect that by checking if its syntax is the left hand side of a member initializer that has object initializer of the right?

**Comment by 333fred** [comment]
Superceded by https://github.com/dotnet/roslyn/pull/49534.

**Pull request closed without merging**